server/tserver/src/main/java/org/apache/accumulo/tserver/scan/NextBatchTask.java

```diff
@@ -89,14 +89,15 @@ public void run() {
       }
     } catch (TooManyFilesException | SampleNotPresentException e) {
       addResult(e);
-    } catch (OutOfMemoryError ome) {
-      Halt.halt("Ran out of memory scanning " + scanSession.extent + " for " + scanSession.client,
-          1);
-      addResult(ome);
-    } catch (Throwable e) {
-      log.warn("exception while scanning tablet "
-          + (scanSession == null ? "(unknown)" : scanSession.extent), e);
+    } catch (IOException | RuntimeException e) {
```

Comment:
This is good because it preserves existing behavior with regard to RTEs, but I suspect that eventually, we'll want to reconsider whether we want RTEs to get sent back to the client, or if we would want them to fall through and trigger a TApplicationException (which is like an HTTP 500 error), so that we don't leak arbitrary exceptions back to the client.